File: packages/jsii-rosetta/test/translations/calls/will_type_deep_structs_directly_if_type_info_is_available.ts
/// fake-from-jsii
interface BaseDeeperStruct {
  a: number;
}

interface DeeperStruct extends BaseDeeperStruct {
  b: number;
}

interface OuterStruct {
  foo: number;
  deeper: DeeperStruct;
}

function foo(x: number, outer: OuterStruct) { }

foo(25, { foo: 3, deeper: {
  a: 1,
  b: 2
});

File: packages/jsii-rosetta/test/translations/calls/will_type_deep_structs_directly_if_type_info_is_available.go
type baseDeeperStruct struct {
	a *f64
}

type deeperStruct struct {
	baseDeeperStruct
	b *f64
}

type outerStruct struct {
	foo *f64
	deeper *deeperStruct
}

func foo(x *f64, outer *outerStruct) {
}

foo(jsii.Number(25), &outerStruct{
	Foo: jsii.Number(3),
	Deeper: &deeperStruct{
		A: jsii.Number(1),
		B: jsii.Number(2),
	},
})
